apr dev update (#123)
irods/irods.org: content/posts/irods-development-update-april-2026.md

Title: iRODS Development Update: April 2026
Date: 2026-04-22 10:00
Author: Kory Draughn
Slug: irods-development-update-april-2026
Status: published

Hello reader,

> The iRODS User Group Meeting is right around the corner. If you're working on cool stuff and would like to give a talk, please reach out to us. Registration for the event can be found at <https://irods.org/ugm2026>. We hope to see you there!

Reviewer: ```suggestion
This year's iRODS User Group Meeting is right around the corner. If you're working on cool stuff and would like to give a talk, please reach out to us. Registration for the event can be found at <https://irods.org/ugm2026>. We hope to see you there!
```

Author: Done.

For this month's update, we're going to start things off with the initial release of the iRODS Policy Composition rule engine plugin. It is released as 0.1.0. Visit <https://irods.org/2026/04/initial-release-of-irods-policy-composition/> to read all about it.

> Work on iRODS 5.1.0 is still going strong. The native implementation of the Logical Quotas rule engine plugin is nearing completion. The hashers used by the server to calculate checksums now grant the developer control over the output format. To support FIPS compliance, we've added configuration options for selecting the hash scheme used to sign zone keys. The DataObjRename API has been updated to honor logical locking. We've fixed some bugs pertaining to group permission expansion in GenQuery2 and the filesystem library. Not only that, but GenQuery2 now exposes columns for determining when metadata is attached to an entity. Lastly, work to make the resource-rebalance operation continue when there are no good replicas is in progress.

Reviewer: ```suggestion
Work on iRODS 5.1.0 is still going strong. The native implementation of the Logical Quotas rule engine plugin is nearing completion. The hashers used by the server to calculate checksums now grant the developer control over the output format. To support FIPS compliance, we've added configuration options for selecting the hash scheme used to sign zone keys. The DataObjRename API has been updated to honor logical locking. We've fixed some bugs pertaining to group permission expansion in GenQuery2 and the filesystem library. Not only that, but GenQuery2 now exposes columns for determining when metadata is attached to an entity. Lastly, work to make the resource-rebalance operation continue when there are no good replicas of a data object is in progress.
```

Author: Done.

For the S3 resource plugin, work to add support for CRC64/NVME trailing checksums on upload is complete and has been merged into upstream. This new feature will be available for iRODS 5.1.0.

Python iRODS client (PRC) 3.3.0 is now available. Users can expect improvements in stability and correctness. For the full set of changes, visit <https://github.com/irods/python-irodsclient/releases/tag/v3.3.0>. Since this release, support for the atomic ACLs API has been implemented. We've also started investigating how to enable automated testing for Microsoft Windows.

> To wrap up, the iRODS Consortium will no longer host quarterly TRiRODS meetings. We apprepriate everyone who participated. Going forward, we'll be offering monthly Community Office Hours, where anyone can drop in and ask questions about iRODS.

Reviewer: ```suggestion
To wrap up, the iRODS Consortium will no longer host quarterly [TRiRODS](https://irods.org/trirods/) meetings. We appreciate everyone who participated over the last 10 years. Going forward, we'll be offering monthly Community Office Hours, where anyone can drop in and ask questions about iRODS and how we can help.
```

Author: Done.
